packages/mobile/src/account/FiatExchange.tsx

```diff
@@ -1,104 +1,125 @@
-import colorsV2 from '@celo/react-components/styles/colors.v2'
+import TextButton from '@celo/react-components/components/TextButton.v2'
+import colors from '@celo/react-components/styles/colors.v2'
+import fontStyles from '@celo/react-components/styles/fonts.v2'
+import variables from '@celo/react-components/styles/variables'
+import { CURRENCIES, CURRENCY_ENUM } from '@celo/utils/src'
+import { useNavigation } from '@react-navigation/native'
 import * as React from 'react'
-import { WithTranslation } from 'react-i18next'
-import { ActivityIndicator, StyleSheet } from 'react-native'
+import { useTranslation, WithTranslation } from 'react-i18next'
+import { ActivityIndicator, StyleSheet, Text, View } from 'react-native'
 import SafeAreaView from 'react-native-safe-area-view'
 import { WebView } from 'react-native-webview'
-import { connect } from 'react-redux'
+import { connect, useSelector } from 'react-redux'
 import { showError } from 'src/alert/actions'
 import { ErrorMessages } from 'src/app/ErrorMessages'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import { ListItem } from 'src/fiatExchanges/ListItem'
 import { Namespaces, withTranslation } from 'src/i18n'
 import { LocalCurrencyCode } from 'src/localCurrency/consts'
+import { useDollarsToLocalAmount, useLocalCurrencyCode } from 'src/localCurrency/hooks'
 import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
 import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { Screens } from 'src/navigator/Screens'
 import { RootState } from 'src/redux/reducers'
+import { stableTokenBalanceSelector } from 'src/stableToken/reducer'
 
-interface State {
-  signedUrl: string
-}
-
-interface StateProps {
-  localCurrency: LocalCurrencyCode
-  account: string | null
-}
-
-const mapStateToProps = (state: RootState): StateProps => {
-  return {
-    localCurrency: getLocalCurrencyCode(state),
-    account: state.web3.account,
-  }
-}
+interface OwnProps {}
+type Props = OwnProps
 
-type Props = StateProps & WithTranslation
-
-const celoCurrencyCode = 'CUSD'
-const signMoonpayFirebaseUrl = 'https://us-central1-celo-org-mobile.cloudfunctions.net/signMoonpay'
-
-async function signMoonpayUrl(account: string, localCurrencyCode: LocalCurrencyCode) {
-  const response = await fetch(signMoonpayFirebaseUrl, {
-    method: 'POST',
-    headers: {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify({
-      currency: celoCurrencyCode,
-      address: account,
-      fiatCurrency: localCurrencyCode,
-    }),
-  })
-  const json = await response.json()
-  return json.url
-}
-
-class FiatExchange extends React.Component<Props, State> {
-  state: State = {
-    signedUrl: '',
+function FiatExchange(props: Props) {
+  const { t } = useTranslation()
+  const navigation = useNavigation()
+  const goToAddFunds = React.useCallback(() => {
+    navigation.navigate(Screens.FiatExchange)
```

Comment:
I'm not sure if this would ever happen, but shouldn't navigation be in the dependencies array?

Reply:
Yeap, it is very much in progress, not really ready to review yet :) But thanks for the comments, gives me a good ideas to improve and hot to continue!